### PR TITLE
feat(progamableApi): CMD options can be passed programatically now

### PR DESCRIPTION
**Pull request by BorntraegerMarc**
Fixes https://github.com/db-migrate/node-db-migrate/issues/553

### CODE REVIEW

**Comment by BorntraegerMarc** [comment]
@wzrdtales The new PR with your suggested changes. Could you maybe release a new version if you think it's good? 😄

**Comment by BorntraegerMarc** [comment]
@wzrdtales and do you know why the CI fails? 😅

File: lib/commands/set-default-argv.js [hunk 1]
```diff
@@ -98,9 +98,10 @@ module.exports = function (internals, isModule) {
       .describe('log-level', 'Set the log-level, for example sql|warn')
       .string('log-level');
   } else {
+    var _internalsArgv = Object.assign(defaultConfig, internals.cmdOptions);
```
Comment (wzrdtales) [inline lib/commands/set-default-argv.js]: we stop using var from now on, since we dropped support prior node 4. Use const here instead.

**Comment by wzrdtales** [comment]
Arggh...

Seems the node team backported some features to node 8

```
The following leaks were detected:SharedArrayBuffer, Atomics
```

**Comment by wzrdtales** [comment]
I pushed out a change that fixes the ci errors.

**Comment by BorntraegerMarc** [comment]
@wzrdtales didn't know that node v4 support was dropped. But I think it's very cool :) Ready for anoter review

**Comment by wzrdtales** [comment]
everything before node 4, node 4 itself is supported though

**Comment by wzrdtales** [comment]
Looks good, please bare with me, I have some meetings now, but will publish ASAP. Ping me should I not publish it before tomorrow morning :)

**Comment by wzrdtales** [comment]
Pushed out. For the next time few hints:

Here you can find a description of the conventions for commits.
https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular

Your second commit would have better suited the refactor type :)

**Comment by wzrdtales** [comment]
+ db-migrate@0.10.6

**Comment by BorntraegerMarc** [comment]
@wzrdtales noted. thanks for the tip :)

**Pull request merged**